File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.js
function solution(s){
    let answer = true;
    const stack = [];
    s = s.split('');
    for(let i=0; i < s.length; i++){
        if(s[i] === "(") {
            stack.push(s[i]);
        } else {
            if(stack.length === 0) stack.push("(");
            else stack.pop();
        }
    }
    
    if(stack.length > 0) answer = false
    else answer = true;
    
    return answer;
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
class Solution {
    boolean solution(String s) {
        boolean answer = true;
        int count = 0;
        String[] arr = s.split("");
        
        for(int i = 0; i < arr.length; i++){
            if(count == 0 && arr[i].equals(")")){
                    return false;
            }
            
            if(arr[i].equals("(")){
                count++;
            } else {
                count--;
            }
        }
        
        return count == 0 ? true : false;
    }
}
